imageio/core/functions.py

```diff
@@ -76,12 +76,55 @@
 
 from __future__ import absolute_import, print_function, division
 
+from numbers import Number
+import re
+
 import numpy as np
 
 from . import Request, RETURN_BYTES
 from .. import formats
 
 
+MEMTEST_DEFAULT_MIM = "256MB"
+MEMTEST_DEFAULT_MVOL = "1GB"
+
+
+mem_re = re.compile(r"^(\d+\.?\d*)\s*([kKMGTPEZY]?i?[bB])?$")
+sizes = {"": 1, None: 1}
+for i, si in enumerate([""] + list("kMGTPEZY")):
+    for b, divisor in [("b", 8), ("B", 1)]:
+        sizes[si + b] = 1000 ** i / divisor
+        sizes[si.upper() + "i" + b] = 1024 ** i / divisor
```

Comment:
You've made lowercase "b" mean bits, but I think we can assume that we always talk about bytes. I'd prefer to make this mechanism completely case insensitive: I think it would not be uncommon for people to write simply "100mb".

As I understand it (but correct me if I'm wrong!), the "MiB" notation explicitly means `1024` (`2**10`) multipliers, whereas for "MB" its can mean either. In practice almost everybody uses the 1024 multiplier, with the exception of hard drive manufacturers. So I'd prefer to always use the 1024 multiplier, and allow both the "MiB" and "MB" notation.

Reply:
My understanding is that there was a period where the SI prefixes meant jumps of 1000, unambiguously. Then it was decided that computers working in binary meant that we may as well use 1024 when talking about computers, which created ambiguity as context-dependent unit prefixes are not a good idea. So in \*hits wikipedia\* 1998, a standard was published giving binary prefixes their own letters, and thereby making SI units Standard again. k unambiguously means 1000, and Ki unambiguously means 1024, according to all major standards organisations and the SI system.

I fully accept that I am more nitpicky than most about this but I personally have strong feelings about maintaining correctness in this regard - in both the prefixes and the b/B. SI units are valuable because they are standardised and unambiguous. If people want to use them incorrectly, they're welcome to: that's the "consenting adults" ethos of python. We don't account for typos in file names either. I don't think the kilo/kibi will make much of a difference to anyone, but I'll add to the docstring to make it clear that proper capitalisation is expected.

For me, this change was about empowering users to write expressive, grok-able code, rather than saving keystrokes.

Reply:
... but I'll also stop it from accepting `"i"` as a unit.

Reply:
> but I personally have strong feelings about maintaining correctness in this regard 

I'm with you on that :)

As for `b` vs `B`, what about only supporting uppercase `B`? Using lowercase `b` will be invalid, because the argument should be given in bytes, not bits. This way we have correctness without user confusion (a quick line in the docs should sort this out).

As for whether `1MB` should/can mean `2**20` bytes or `10**6`, I'm still not sure. If I right-click a file on Windows, it may say `1 MB (1048576 bytes)`. On Ubuntu `1 MB (1000000 bytes). This page is also not clear https://en.wikipedia.org/wiki/Byte#Unit_multiples. That said, the difference will be low in practice, and I'd be happy to go with your approach.

Reply:
> On Ubuntu `1 MB (1000000 bytes)`

Oh boy is there drama on that: https://wiki.ubuntu.com/UnitsPolicy From the looks of it, [it varies by program](https://askubuntu.com/a/897397/439260). You're right that there is more ambiguity in the general ecosystem than I optimistically implied, though.

I'm happy to drop `'b'` though, as it'll remove the most likely pain point and as you said before, few people think in bits (I was trying to imagine a situation where someone might be thinking in terms of data transmission speed, but came up short).